pkg/log/logshipper/logshipper.go

```diff
@@ -64,24 +64,30 @@ func (ls *LogShipper) Ping() {
 	ls.sender.authtoken = string(token)
 
 	shouldEnable := ls.knapsack.LogIngestServerURL() != ""
-	parsedUrl, err := url.Parse(ls.knapsack.LogIngestServerURL())
-	if err != nil || parsedUrl.String() == "" {
-		// If we have a bad endpoint, just disable for now.
-		// It will get renabled when control server sends a
-		// valid endpoint.
-		shouldEnable = false
-		level.Debug(ls.baseLogger).Log(
-			"msg", "error parsing log ingest server url, shipping disabled",
-			"err", err,
-			"log_ingest_url", ls.knapsack.LogIngestServerURL(),
-		)
+
+	// Only attempt to parse a URL if we have one
+	if shouldEnable {
+		parsedUrl, err := url.Parse(ls.knapsack.LogIngestServerURL())
+		if err != nil {
+			// If we have a bad endpoint, just disable for now.
+			// It will get renabled when control server sends a
+			// valid endpoint.
+			shouldEnable = false
+			level.Debug(ls.baseLogger).Log(
+				"msg", "error parsing log ingest server url, shipping disabled",
+				"err", err,
+				"log_ingest_url", ls.knapsack.LogIngestServerURL(),
+			)
+		} else if parsedUrl != nil {
```

Comment:
`|| u.String() == ""` maybe? But that feels kinda gross

Reply:
I didn't want to do that because `parsedUrl.String()` will panic when the parsedUrl is nil, and I believe it's possible for parsing to return nil without also returning an error per the docs:

> Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities. 

So this is a safer check

Reply:
Oh sorry I misread your comment!